### PR TITLE
Implement Plantrendering

### PR DESCRIPTION
**Pull request by NilsAffing**
Plants can be rendered
Sticks are rendered correctly

### CODE REVIEW

File: client/src/view/plant_view.rs [hunk 1]
```diff
@@ -22,30 +25,30 @@ impl PlantView {
                                  facade: &F)
                                  -> Self {
         // FIXME handle other plant types
-        let mut verts = 0;
+        // let verts = 0;
```
Comment (jonas-schievink) [inline client/src/view/plant_view.rs]: Remove this comment

File: client/src/view/plant_dummy.vert [hunk 2]
```diff
@@ -2,13 +2,22 @@
 
 in vec3 position;
 in vec3 color;
+in vec3 normal;
+
 out vec3 x_color;
+out vec3 surfaceNormal;
+out vec3 toLight;
```
Comment (LukasKalbertodt) [inline client/src/view/plant_dummy.vert]: This is currently a vertex attribute but it holds the same value for all vertices. Use a uniform variable instead that you can access in both shaders.

File: client/src/world/chunk_std.frag [hunk 3]
```diff
@@ -7,6 +7,6 @@ in vec3 surfaceNormal;
 out vec3 color;
 
 void main() {
-    float diffuse = max(0.0, dot(toLight, surfaceNormal));
+    float diffuse = max(0.0, dot(normalize(toLight), normalize(surfaceNormal)));
```
Comment (jonas-schievink) [inline client/src/world/chunk_std.frag]: Is this necessary?

Comment (NilsAffing) [inline client/src/world/chunk_std.frag]: Yes, because we calculate the normals dynamically and the length of these vectors are not 1.

Comment (NilsAffing) [inline client/src/world/chunk_std.frag]: Never mind ;)

**Pull request merged**